src/tests/ftest/io/nvme_pool_capacity.py

```diff
@@ -0,0 +1,268 @@
+        manager = Mpirun(ior_cmd, mpitype="mpich")
+        manager.job.daos_cont.update(container_info
+                                     ["{}{}{}".format(oclass,
+                                                      api,
+                                                      test[2])])
+        env = ior_cmd.get_default_env(str(manager))
+        manager.setup_command(env, self.hostfile_clients,
+                              processes)
+
+        # run IOR Command
+        try:
+            manager.run()
+        except CommandFailure as _error:
+            results.put("FAIL")
+
+    def test_create_delete(self, num_pool=2, num_cont=5, total_count=100,
+                           scm_size=100000000000, nvme_size=300000000000):
+        """
+        Test Description:
+            This method is used to create/delete pools
+            for a long run. It verifies the NVME free space
+            during this process.
+            Args:
+                num_pool (int): Total pools for running test
+                num_cont (int): Total containers created on each pool
+                total_count (int): Total times the test is run in a loop
+                scm_size (int): SCM size used in the testing
+                nvme_size (int): NVME size used in the testing
+            Returns:
+                None
+        """
+        pool = {}
+        cont = {}
+
+        for loop_count in range(0, total_count):
+            self.log.info("Running test %s", loop_count)
+            for val in range(0, num_pool):
+                pool[val] = TestPool(self.context,
+                                     dmg_command=self.get_dmg_command())
+                pool[val].get_params(self)
+                # Split total SCM and NVME size for creating multiple pools.
+                temp = int(float(scm_size) / num_pool)
+                pool[val].scm_size.update(str(temp))
+                temp = int(float(nvme_size) / num_pool)
+                pool[val].nvme_size.update(str(temp))
+                pool[val].create()
+                self.pool = pool[val]
+                display_string = "pool{} space at the Beginning".format(val)
+                self.pool.display_pool_daos_space(display_string)
+                nvme_size_begin = self.pool.get_pool_free_space("NVME")
+                for cont_val in range(0, num_cont):
+                    cont[cont_val] = TestContainer(pool[val])
+            m_leak = 0
+            for val in range(0, num_pool):
+                display_string = "Pool{} space at the End".format(val)
+                self.pool = pool[val]
+                self.pool.display_pool_daos_space(display_string)
+                nvme_size_end = self.pool.get_pool_free_space("NVME")
+                pool[val].destroy()
+                if (nvme_size_begin != nvme_size_end) and (m_leak == 0):
+                    m_leak = val
+            # After destroying pools, check memory leak for each test loop.
+            if m_leak != 0:
```

Comment:
Since val starts at 0, can we detect memory leak that occurred in the first iteration in above for loop?

Reply:
Good catch... Will increment the value by one... Thanks.